code/Control/UserDefinedFormController.php

```diff
@@ -511,6 +533,250 @@ public function finished()
+        $form->setFormAction("UserDefinedFormController/confirmfolder");
+
+
+        if (!$editableFormField instanceof EditableFileField) {
+            $editableFormField->setClassName(EditableFileField::class);
+            $editableFormField->write();
+        }
+        $treeView->setSchemaData([
+            'data' => [
+                'urlTree' => "admin/pages/edit/EditForm/$userFormID/field/Fields/item/$id/ItemEditForm/field/FolderID/tree"
+            ]
+        ]);
+
+
+        // create the schema response
+        $parts = $this->getRequest()->getHeader(LeftAndMain::SCHEMA_HEADER);
+        $schemaID = $this->getRequest()->getURL();
+        $data = FormSchema::singleton()
+            ->getMultipartSchema($parts, $schemaID, $form);
+
+        // return the schema response
+        $response = new HTTPResponse(json_encode($data));
+        $response->addHeader('Content-Type', 'application/json');
+        return $response;
+    }
+
+    /**
+     * Sets the selected folder as the upload folder for an EditableFileField
+     * @return HTTPResponse
+     * @throws ValidationException
+     */
+    public function confirmfolder()
+    {
+        $request = $this->getRequest();
+
+        // retrieve the EditableFileField
+        $id = $request->requestVar('ID');
+        /** @var EditableFileField $editableFileField */
+        $editableFormField = EditableFormField::get()->byID($id);
+        if (!$editableFormField) {
+            $editableFormField = Versioned::get_by_stage(EditableFormField::class, Versioned::DRAFT)->byID($id);
+        }
+        // change the class if it is incorrect
+        if (!$editableFormField instanceof EditableFileField) {
+            $editableFormField->setClassName(EditableFileField::class);
+        }
+        $editableFileField = $editableFormField;
+
+        if (!$editableFileField->canView()) {
+            throw new PermissionFailureException();
+        }
+
+        // check if we're creating a new folder or using an existing folder
+        $option = $request->requestVar('FolderOptions');
+        if ($option == 'existing') {
+            // set existing folder
+            $folderID = $request->requestVar('FolderID');
+            $folder = Folder::get()->byID($folderID);
+        } else {
+            // create the folder
+            $createFolder = $this->config()->get('form_submissions_folder') . '/' . $request->requestVar('CreateFolder');
+            $formSubmissionsFolderExists = !!Folder::find($this->config()->get('form_submissions_folder'));
+            $folder = Folder::find_or_make($createFolder);
+
+            // If this is the first time we create the form submission folder
+            if (!$formSubmissionsFolderExists) {
+                // ensure the FormSubmissions folder is only accessible to Administrators
+                $formSubmissionsFolder = Folder::find($this->config()->get('form_submissions_folder'));
```

Comment:
Shouldn't this be find_or_make?

Reply:
The form submission folder will be created just before.